starport/pkg/field/field_test.go

```diff
@@ -0,0 +1,91 @@
+package field_test
+
+import (
+	"errors"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/starport/starport/pkg/field"
+	"github.com/tendermint/starport/starport/pkg/multiformatname"
+	"testing"
+)
+
+var (
+	noCheck       = func(string) error { return nil }
+	alwaysInvalid = func(string) error { return errors.New("invalid name") }
+)
+
+type testCases struct {
+	provided []string
+	expected []field.Field
+}
+
+func TestParseFields(t *testing.T) {
+	cases := testCases{
+		provided: []string{
+			"foo",
+			"bar:string",
+			"fooBar:bool",
+			"bar-foo:int",
+			"foo_foo:uint",
+		},
+		expected: []field.Field{
+			{
+				Datatype:     "string",
+				DatatypeName: "string",
+			},
+			{
+				Datatype:     "string",
+				DatatypeName: "string",
+			},
+			{
+				Datatype:     "bool",
+				DatatypeName: "bool",
+			},
+			{
+				Datatype:     "int32",
+				DatatypeName: "int",
+			},
+			{
+				Datatype:     "uint64",
+				DatatypeName: "uint",
+			},
+		},
+	}
+	cases.expected[0].Name, _ = multiformatname.NewMultiFormatName("foo")
```

Comment:
can we refactor this to execute each test over a loop like a table test?

Reply:
I made this part a bit cleaner but I'm not sure it it what you meant. This is already a table test but we have a single test that match an expected with an actual array